File: algorithms/selection-sort/selection-sort.go
package selectionSort

func selection_sort(L []int) []int {
	for i := 0; i < len(L); i++ {
		min := L[i]
		for j := i; j < len(L); j++ {
			if L[j] < min {
				tmp := min
				min = L[j]
				L[j] = tmp
			}
		}
		L[i] = min
	}
	return L
}

File: algorithms/selection-sort/selection-sort.c
#include <stdlib.h>
#include <stdio.h>

void selection_sort(int* L) {
    int length = sizeof(L)/sizeof(int);
    for(int i = 0; i < length; i++) {
        int min = L[i];
        for(int j=i; j< length; j++) {
            if(L[j] < min) {
                int tmp = min;
                min = L[j];
                L[j] = tmp;
            }
        }
        L[i] = min;
    }
}
int main(int argc, char** argv) {
    int nums[] = {9,8,6,5,4,3,2,1};
    selection_sort(nums);
    for(int i=0; i<sizeof(nums)/sizeof(int); i++) {
        printf("%d, ", nums[i]);
    } 
    printf("\n");
    return 0;
}
